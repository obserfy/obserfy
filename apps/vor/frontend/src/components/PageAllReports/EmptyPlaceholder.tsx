import { Trans } from "@lingui/macro"

import { Button, Flex, Text } from "theme-ui"
import { ReactComponent as EmptyIllustration } from "../../images/report-illustration.svg"
import { NEW_REPORT_URL } from "../../routes"
import { Link } from "../Link/Link"

const EmptyPlaceholder = () => (
  <Flex
    sx={{
      flexDirection: ["column-reverse", "column-reverse", "row"],
      justifyContent: ["flex-end", "center"],
      alignItems: "center",
      width: "100%",
      minHeight: "100vh",
    }}
    pt={[4, 0]}
  >
    <Flex
      m={3}
      mr={[3, 3, 5]}
      sx={{
        maxWidth: 400,
        flexDirection: "column",
        alignItems: ["center", "center", "flex-start"],
      }}
    >
      <Text
        mb={3}
        sx={{
          fontSize: [4, 5],
          fontWeight: "bold",
          textAlign: ["center", "center", "inherit"],
        }}
      >
        <Trans>Write progress reports</Trans>
      </Text>
      <Text
        mb={3}
        sx={{
          textAlign: ["center", "center", "inherit"],
          fontSize: 2,
          color: "textMediumEmphasis",
        }}
      >
        <Trans>
          Easily write reports for parents with all the data that you need in
          one place.
        </Trans>
      </Text>

      <Link to={NEW_REPORT_URL}>
        <Button>
          <Trans>Start your first report</Trans>
        </Button>
      </Link>
    </Flex>

    <EmptyIllustration
      sx={{
        pt: [4, 0],
        width: [240, 320],
      }}
    />
  </Flex>
)

export default EmptyPlaceholder
