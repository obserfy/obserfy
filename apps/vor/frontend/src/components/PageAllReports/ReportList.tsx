import { StaticImage } from "gatsby-plugin-image"
import { FC } from "react"
import { Box, Button, Flex, Text } from "theme-ui"
import { ProgressReport } from "../../__generated__/models"
import { borderBottom, borderFull, borderTop } from "../../border"
import dayjs from "../../dayjs"
import { ReactComponent as PlusIcon } from "../../icons/plus.svg"
import { MANAGE_REPORT_URL, NEW_REPORT_URL } from "../../routes"
import Icon from "../Icon/Icon"
import { Link } from "../Link/Link"

const ReportList: FC<{ reports: ProgressReport[] }> = ({ reports }) => {
  return (
    <Box>
      <Box
        sx={{
          ...borderBottom,
          position: "relative",
          top: 0,
          background:
            "linear-gradient(52deg, rgba(34,90,195,1) 0%, rgba(45,253,130,1) 100%)",
        }}
      >
        <StaticImage
          alt=""
          src="../../../static/background/progress-report.jpeg"
          placeholder="blurred"
          sx={{
            opacity: 0.2,
            position: "absolute",
            top: 0,
            bottom: 0,
            right: 0,
            left: 0,
            zIndex: 0,
          }}
        />

        <Flex
          px={[3, 3, 3, 0]}
          mx="auto"
          sx={{
            position: "relative",
            alignItems: "flex-end",
            maxWidth: "maxWidth.lg",
            zIndex: 1,
            height: 160,
          }}
        >
          <Text
            as="h1"
            sx={{
              fontSize: 5,
              fontWeight: "bold",
              color: "white",
              lineHeight: 1.1,
              fontFamily: "display",
            }}
            mb={3}
            pb={2}
          >
            Progress Reports
          </Text>

          <Link
            to={NEW_REPORT_URL}
            sx={{
              flexShrink: 0,
              ml: "auto",
              mb: -19,
            }}
          >
            <Button variant="secondary" sx={{ borderRadius: "circle" }}>
              <Icon as={PlusIcon} sx={{ fill: "onSecondary" }} mr={1} />
              New Report
            </Button>
          </Link>
        </Flex>
      </Box>

      <Box
        mt={4}
        mx={[0, 3, 3, "auto"]}
        sx={{
          ...borderFull,
          borderLeftStyle: ["none", "solid"],
          borderRightStyle: ["none", "solid"],
          borderRadius: [0, "default"],
          backgroundColor: "surface",
          maxWidth: "maxWidth.lg",
          overflow: "hidden",
        }}
      >
        <Flex p={3} sx={{ backgroundColor: "darkSurface" }}>
          <Text color="textMediumEmphasis" sx={{ width: "65%" }}>
            Title
          </Text>
          <Text
            color="textMediumEmphasis"
            sx={{ width: "20%", display: ["none", "none", "block"] }}
          >
            Start Date
          </Text>
          <Text color="textMediumEmphasis" sx={{ width: "5%" }}>
            Status
          </Text>
        </Flex>
        {reports.map((report) => (
          <Link
            to={MANAGE_REPORT_URL(report.id)}
            key={report.id}
            sx={{
              p: 3,
              ...borderTop,
              display: "flex",
              width: "100%",
              alignItems: "center",
              flexWrap: "wrap",
              "&:hover": {
                backgroundColor: "primaryLightest",
              },
            }}
          >
            <Text sx={{ width: "65%" }}>{report.title}</Text>
            <Text
              sx={{ width: "20%", display: ["none", "none", "block"] }}
              color="textMediumEmphasis"
            >
              {dayjs(report.periodStart).format("DD MMM YYYY")}
            </Text>
            {report.published ? (
              <Flex
                px={2}
                sx={{
                  ...borderFull,
                  borderRadius: "circle",
                  alignItems: "baseline",
                  backgroundColor: "surface",
                }}
              >
                <Box
                  mb={1}
                  mr={2}
                  sx={{
                    backgroundColor: "primaryDark",
                    width: "8px",
                    height: "8px",
                    borderRadius: "circle",
                    display: ["none", "block"],
                  }}
                />
                <Text sx={{ fontSize: 0, width: "10%", color: "primaryDark" }}>
                  Published
                </Text>
              </Flex>
            ) : (
              <Flex
                px={2}
                sx={{
                  ...borderFull,
                  borderRadius: "circle",
                  alignItems: "baseline",
                  backgroundColor: "surface",
                }}
              >
                <Box
                  mb={1}
                  mr={2}
                  sx={{
                    width: "8px",
                    height: "8px",
                    backgroundColor: "warning",
                    borderRadius: "circle",
                    display: ["none", "block"],
                  }}
                />
                <Text sx={{ fontSize: 0, width: "15%", color: "warning" }}>
                  Unpublished
                </Text>
              </Flex>
            )}

            <Text
              color="textMediumEmphasis"
              mt={1}
              sx={{ width: "100%", display: ["block", "block", "none"] }}
            >
              {dayjs(report.periodStart).format("DD MMM YYYY")}
            </Text>
          </Link>
        ))}
      </Box>
    </Box>
  )
}

export default ReportList
