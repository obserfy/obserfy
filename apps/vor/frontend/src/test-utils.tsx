/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
