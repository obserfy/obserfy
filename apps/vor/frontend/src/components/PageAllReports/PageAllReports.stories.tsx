import { Meta, Story } from "@storybook/react"
import PageAllReports, { PageAllReportsProps } from "./PageAllReports"

export default {
  title: "Core/PageAllReports",
  component: PageAllReports,
  parameters: {
    componentSubtitle: "Just a simple PageAllReports",
  },
} as Meta

const Template: Story<PageAllReportsProps> = (args) => (
  <PageAllReports {...args} />
)

export const Default = Template.bind({})
Default.args = {}
