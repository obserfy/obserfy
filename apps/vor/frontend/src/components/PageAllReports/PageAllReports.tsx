import { FC } from "react"
import { Box } from "theme-ui"
import { ProgressReport } from "../../__generated__/models"
import useGetProgressReports from "../../hooks/api/useGetProgressReports"
import EmptyPlaceholder from "./EmptyPlaceholder"
import ReportLists from "./ReportList"

export interface PageAllReportsProps {}

const PageAllReports: FC<PageAllReportsProps> = () => {
  const reports = useGetProgressReports()

  if (reports.isLoading) return <div>Loading</div>

  if (reports.isSuccess) return <Success reports={reports.data} />

  return <Box p={3}>Error</Box>
}

const Success: FC<{ reports: ProgressReport[] }> = ({ reports }) => {
  if (reports.length === 0) {
    return <EmptyPlaceholder />
  }
  return <ReportLists reports={reports} />
}

export default PageAllReports
