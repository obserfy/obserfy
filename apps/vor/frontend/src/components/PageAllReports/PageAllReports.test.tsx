import { render } from "../../test-utils"
import PageAllReports from "./PageAllReports"

describe("PageAllReports", () => {
  it("should render correctly", () => {
    const { container } = render(<PageAllReports />)
    expect(container).toMatchSnapshot()
  })
})
