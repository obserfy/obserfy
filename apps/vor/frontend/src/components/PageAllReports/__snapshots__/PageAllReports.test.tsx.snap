// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageAllReports should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

<div>
  <div
    class="emotion-0"
  >
    <div>
      Loading
    </div>
  </div>
</div>
`;
